commit 16175a796d061833aacfbd9672235f2d2725df65
Author: Avi Kivity <avi@redhat.com>
Date:   Mon Mar 23 22:13:44 2009 +0200

    KVM: VMX: Don't allow uninhibited access to EFER on i386
    
    vmx_set_msr() does not allow i386 guests to touch EFER, but they can still
    do so through the default: label in the switch.  If they set EFER_LME, they
    can oops the host.
    
    Fix by having EFER access through the normal channel (which will check for
    EFER_LME) even on i386.
    
    Reported-and-tested-by: Benjamin Gilbert <bgilbert@cs.cmu.edu>
    Cc: stable@kernel.org
    Signed-off-by: Avi Kivity <avi@redhat.com>

Adjusted to apply to Debian's 2.6.24 by dann frazier <dannf@debian.org>

diff -urpN a/drivers/kvm/vmx.c b/drivers/kvm/vmx.c
--- a/drivers/kvm/vmx.c	2008-01-24 15:58:37.000000000 -0700
+++ b/drivers/kvm/vmx.c	2009-04-08 22:46:00.000000000 -0600
@@ -709,12 +709,12 @@ static int vmx_set_msr(struct kvm_vcpu *
 	int ret = 0;
 
 	switch (msr_index) {
-#ifdef CONFIG_X86_64
 	case MSR_EFER:
 		ret = kvm_set_msr_common(vcpu, msr_index, data);
 		if (vmx->host_state.loaded)
 			load_transition_efer(vmx);
 		break;
+#ifdef CONFIG_X86_64
 	case MSR_FS_BASE:
 		vmcs_writel(GUEST_FS_BASE, data);
 		break;
